### PR TITLE
Add eval and explain functionalities to UI and APIs

### PR DESCRIPTION
**Pull request by am357**
*Description of changes:*

- Adds query evaluation and `Explain` to the playground using `partiql-lang-rust` `0.2.0` new features.
- In addition it adds `/explain` and `/eval` REST APIs for `Explain` and evaluation functionalities.

### UI Demo
Feb-15-2023 16-30-00

### CLI Demo
Feb-16-2023 08-58-26

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by yliuuuu — COMMENTED** [review]
Pull down and tested. Can confirm the submodule build is now successful.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,46 @@
 # PartiQL Playground (Proof of Concept)
 
-PartiQL Playground is intended to provide the required code for enabling execution of PartiQL queries on web.
+PartiQL Playground provides a functionality to execute PartiQL on client on web environment. It also provides
```
Comment (alancai98) [inline README.md:3]: nit: wording

```suggestion
PartiQL Playground provides the functionality to execute PartiQL in a web environment. It also provides
```

File: src/lib.rs [hunk 2]
```diff
@@ -1,14 +1,96 @@
...
+        }
+    }
+}
+
+pub(crate) fn parse(statement: &str) -> ParserResult {
+    Parser::default().parse(statement)
+}
+
+fn eval(p: &Parsed, env: &str) -> partiql_value::Value {
+    let lowered = lower(&p);
```
Comment (alancai98) [inline src/lib.rs:58]: Copying the clippy warning here:

```
this expression creates a reference which is immediately dereferenced by the compiler

warning: this expression creates a reference which is immediately dereferenced by the compiler
  --> src/lib.rs:58:25
   |
58 |     let lowered = lower(&p);
   |                         ^^ help: change this to: `p`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
   = note: `#[warn(clippy::needless_borrow)]` on by default
```

File: src/lib.rs [hunk 3]
```diff
@@ -1,14 +1,96 @@
...
+    let env_as_value = parse_ion(env);
+    let bindings: MapBindings<partiql_value::Value> = MapBindings::from(env_as_value);
+    evaluate(lowered, bindings)
+}
+
+/// Creates a logical plan for the given query using and returns the json serialized string.
```
Comment (alancai98) [inline src/lib.rs:64]:
```suggestion
/// Creates a logical plan for the given query and returns the json serialized string.
```

File: src/lib.rs [hunk 4]
```diff
@@ -1,14 +1,96 @@
...
+#[wasm_bindgen]
+pub fn explain_as_json(statement: &str) -> String {
+    let parsed = parse(statement);
+    let lowered = lower(&parsed
+        .expect("Error in unwrapping json serde"));
+    serde_json::to_string_pretty(&format!("{:?}", lowered))
+        .expect("Error in unwrapping json serde")
+}
+
+/// Creates a logical plan for the given query using and returns the output string.
```
Comment (alancai98) [inline src/lib.rs:74]:
```suggestion
/// Creates a logical plan for the given query and returns the output string.
```

File: src/index.html [hunk 5]
```diff
@@ -69,6 +74,9 @@ <h5 class="modal-title" id="exportModalLabel">Export</h5>
     <div class="row" id="edit-area">
         <div class="col-12" id="editor"></div>
     </div>
+    <div class="row mt-3" id="eval-result-div">
+        <pre id="eval-result" class="form-control"></pre>
+    </div>
     <div class="row mt-3" id="env-div">
         <textarea id="env-data" class="form-control" placeholder="Environment data"></textarea>
```
Comment (alancai98) [inline src/index.html:81]: Maybe as a future TODO/GH issue, could include an example environment to show the structure of the environment data

**Review by alancai98 — APPROVED** [review]
Great work getting the playground working with `partiql-lang-rust` v0.2.0! Left some minor comments related to wording and a suggestion

File: src/index.html (hunk 5 above)
Comment (am357) [reply src/index.html:81]: @alancai98 you mean as part of the environment text area pre-filled or somewhere else?

Comment (am357) [reply src/index.html:84]: https://github.com/partiql/partiql-rust-playground/issues/7

Comment (alancai98) [reply src/index.html:84]: Could be pre-filled or if there were more text instructions on the playground page, we could include the sample environment there.

Something similar to what's done for:
- Amazon Ion: https://amazon-ion.github.io/ion-docs/index.html
- Amazon Ion Schema: https://amazon-ion.github.io/ion-schema/sandbox

**Pull request merged**